File: udemy/ds-and-a/csharp-solutions/DSAndA.Test/LinkedLists/Assignment5/DeleteFromMiddle.Test.cs
using DSAndA.LinkedLists;
using Xunit;

namespace DSAndA.Test.LinkedLists.Assignment5
{
    public class DeleteFromMiddleTestFixture
    {
        [Fact]
        public void DeleteFromMiddleTest1()
        {
            LinkedList<string> linkedList = new LinkedList<string>();
            linkedList.InsertAtHead("z");
            linkedList.InsertAtHead("y");
            linkedList.InsertAtHead("x");
            linkedList.InsertAtHead("w");
            linkedList.InsertAtHead("v");
            linkedList.InsertAtHead("u");

            linkedList.DeleteFromMiddle(linkedList.Search("v"));

            Assert.Equal(expected: "u,w,x,y,z,", actual: linkedList.ToString());
        }
    }
}

File: udemy/ds-and-a/javascript-solutions/tests/LinkedLists/Assignment5/DeleteFromMiddle.test.js
const LinkedList = require("../../../src/LinkedLists/LinkedList");

test('Delete node from somewhere in the middle of a singly linked list', () => {
    let linkedList = new LinkedList();
    linkedList.insertAtHead('z');
    linkedList.insertAtHead('y');
    linkedList.insertAtHead('x');
    linkedList.insertAtHead('w');
    linkedList.insertAtHead('v');
    linkedList.insertAtHead('u');

    expect(linkedList.toString()).toBe('u, v, w, x, y, z');

    linkedList.deleteFromMiddle('v');

    expect(linkedList.toString()).toBe('u, w, x, y, z');
});